### PR TITLE
VIA Support for Boardwalk

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

[title]

## Types of Changes

- [x] Enhancement/optimization

## Checklist

- [x] My code follows the code style of this project.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardwalk/config.h [hunk 1]
```diff
@@ -21,8 +21,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define VENDOR_ID       0xCDCD
 #define PRODUCT_ID      0x5337
 #define DEVICE_VER      0x0001
-#define MANUFACTURER    shens
-#define PRODUCT         Boardwalk
+#define MANUFACTURER    shensmobile
+#define PRODUCT         shensmobile Boardwalk
```
Comment (fauxpark) [inline keyboards/boardwalk/config.h]:
```suggestion
#define PRODUCT         Boardwalk
```

File: keyboards/boardwalk/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1 @@
+VIA_ENABLE = yes
```
Comment (mechmerlin) [inline keyboards/boardwalk/keymaps/via/rules.mk]: Let's add LTO_ENABLE here too!

```suggestion
LTO_ENABLE = yes
```

File: keyboards/boardwalk/config.h (hunk 1 above)
Comment (noroadsleft) [reply keyboards/boardwalk/config.h]: I changed this based on a suggestion from the VIA docs (linked in the commit body), but I can revert this if you insist.

Also did you want the whole changeset reverted, or just the Product line?

Comment (fauxpark) [reply keyboards/boardwalk/config.h]: Just `PRODUCT`. IMO it doesn't make sense to combine them as that's exactly what `MANUFACTURER` is for.

Comment (Wilba6582) [reply keyboards/boardwalk/config.h]: But this is not what Windows shows in the devices, it only shows `PRODUCT` so I suggest making `PRODUCT` the full name you want shown, which may be the make and model.

Comment (fauxpark) [reply keyboards/boardwalk/config.h]: That's just Windows though - the QMK Toolbox will display "shensmobile shensmobile Boardwalk" 👎

Comment (noroadsleft) [reply keyboards/boardwalk/config.h]: I'm feeling like this is a problem better solved in QMK Toolbox / VIA Configurator. Windows is gonna Windows.

**Pull request merged**